### PR TITLE
chore: renames L2ToL1Message to Child/ParentChain

### LINKED ISSUE #342: Update language in `message/L2ToL1Message`

### PR DESCRIPTION
**Pull request by douglance**
closes #342

### CODE REVIEW

File: tests/integration/eth.test.ts
Comment (fionnachan) [inline tests/integration/eth.test.ts:1]: i know that we have future plans of removing `as L2ToL1Message`
but i don't see why we need to use `as ` in a test file at all
we can update this now?

File: src/lib/message/L2ToL1Message.ts [hunk 1]
```diff
@@ -237,81 +246,82 @@ export class L2ToL1MessageReader extends L2ToL1Message {
...
   /**
-   * Executes the L2ToL1Message on L1.
+   * Executes the ChildToParentMessage on ParentChain.
    * Will throw an error if the outbox entry has not been created, which happens when the
    * corresponding assertion is confirmed.
    * @returns
    */
   public async execute(
-    l2Provider: Provider,
+    chainProvider: Provider,
```
Comment (fionnachan) [inline src/lib/message/L2ToL1Message.ts:320]: i think this should be `childChainProvider` to be more explicit

File: src/lib/message/L2ToL1Message.ts [hunk 2]
```diff
@@ -237,81 +246,82 @@ export class L2ToL1MessageReader extends L2ToL1Message {
...
+   * Estimates the ParentChain block number in which this Chain to ParentChain tx will be available for execution.
    * If the message can or already has been executed, this returns null
-   * @param l2Provider
-   * @returns expected L1 block number where the L2 to L1 message will be executable. Returns null if the message can or already has been executed
+   * @param chainProvider
+   * @returns expected ParentChain block number where the Chain to ParentChain message will be executable. Returns null if the message can or already has been executed
    */
   public async getFirstExecutableBlock(
-    l2Provider: Provider
+    chainProvider: Provider
```
Comment (fionnachan) [inline src/lib/message/L2ToL1Message.ts:268]: same question regarding `chain` vs `childChain` in naming

File: src/lib/message/L2ToL1Message.ts [hunk 3]
```diff
@@ -188,45 +192,50 @@ export class L2ToL1Message {
...
+      this.nitroReader = new nitro.L2ToL1MessageReaderNitro(
+        parentChainProvider,
+        event
+      )
     }
   }
 
   public async getOutboxProof(
-    l2Provider: Provider
+    chainProvider: Provider
```
Comment (fionnachan) [inline src/lib/message/L2ToL1Message.ts:221]: i see that some are `childChainProvider` on this file and some are `chainProvider`
i prefer the former for all names

**Review by fionnachan — APPROVED** [review]

**Review by yahgwai — APPROVED** [review]

**Pull request merged**